gk/main.c

```diff
@@ -1988,18 +2108,44 @@ process_pkts_back(uint16_t port_back, uint16_t port_front,
 			continue;
 		}
 
-		if (unlikely((packet.flow.proto == RTE_ETHER_TYPE_IPV4 &&
+		if (unlikely((packet->flow.proto == RTE_ETHER_TYPE_IPV4 &&
 				!ipv4_configured_back) ||
-				(packet.flow.proto == RTE_ETHER_TYPE_IPV6 &&
+				(packet->flow.proto == RTE_ETHER_TYPE_IPV6 &&
 				!ipv6_configured_back))) {
 			drop_packet_back(pkt, instance);
 			continue;
 		}
 
-		fib = look_up_fib(&gk_conf->lpm_tbl, &packet.flow);
+		if (packet->flow.proto == RTE_ETHER_TYPE_IPV4) {
+			lpm_lookup_pos[num_lpm_lookups] = num_ip_flows;
+			flows[num_lpm_lookups] = &packet->flow;
+			num_lpm_lookups++;
+		} else {
+			lpm6_lookup_pos[num_lpm6_lookups] = num_ip_flows;
+			flows6[num_lpm6_lookups] = &packet->flow;
+			num_lpm6_lookups++;
+		}
+
+		num_ip_flows++;
+	}
+
+	lookup_fib_bulk(&gk_conf->lpm_tbl, flows, num_lpm_lookups, fibs);
+	lookup_fib6_bulk(&gk_conf->lpm_tbl, flows6, num_lpm6_lookups, fibs6);
+
+	for (i = 0; i < num_lpm_lookups; i++) {
+		int fidx = lpm_lookup_pos[i];
+
+		process_fib(&pkt_arr[fidx], fibs[i], &num_tx, tx_bufs,
```

Comment:
Does `process_fib()` account for the case in which there's no fib?

Reply:
Yes, if there is no fib for ``i-th`` flow, then ``fibs[i]`` is ``NULL``. The packet will be classified by ACL rules.